### PR TITLE
doc: explanation of how libraries depend on one another

### PR DESCRIPTION
**Pull request by mattkae**
Check out https://canonical-mir--3469.com.readthedocs.build/3469/explanation/libraries/

### CODE REVIEW

File: doc/sphinx/explanation/libraries.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  nothing else depends on them
+- `mirserver` depends on the most libraries as it is the brains of the operation
+- `mircore` and `mircommon` provide common functionalities for every other
+  library in the system
+- The input and graphics platforms (represented as `input_X` and `graphics_X` in
+  the diagram) are dynamically loaded by `mirserver` when the engine starts up33333
+
+## Private Static Libraries
+Mir builds a number of static libraries that are linked to internally by
+`mirserver`. These libraries are:
```
Comment (AlanGriffiths) [inline doc/sphinx/explanation/libraries.md:55]: Your list includes some of the "object" libraries. I think there's only one static library linked by mirserver (kms_utils).

File: doc/sphinx/explanation/libraries.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+In the following diagram, each arrow denotes that the library at the start of
+the arrow depends on the library at the end of the arrow.
+
+```{mermaid} libraries.mmd
+```
+
+There are a few things of note in this diagram:
+
+- `miral` and `miroil` are intended to be the primary user-facing libraries as
+  nothing else depends on them
```
Comment (AlanGriffiths) [inline doc/sphinx/explanation/libraries.md:46]: It is `miral` and `mircore` that are intended to be the primary user-facing libraries.

Miroil needs special mention as a "compatibility layer" for Lomiri that is not intended for general consumption

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Great start!

Personally, I'd drop the section about object libraries as it would need more work I don't think there's enough benefit to justify that

File: doc/sphinx/explanation/libraries.mmd [hunk 3]
```diff
@@ -0,0 +1,18 @@
+flowchart
+    miral ---> mirserver
+    miral ---> mircommon
```
Comment (AlanGriffiths) [inline doc/sphinx/explanation/libraries.mmd:3]: Are you intentionally omitting `miral ---> mircore` to simplify the diagram?

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]

File: doc/sphinx/explanation/libraries.mmd (hunk 3 above)
Comment (mattkae) [reply doc/sphinx/explanation/libraries.mmd:3]: Nope! Just forgot it

**Review by AlanGriffiths — APPROVED** [review]
LGTM

File: doc/sphinx/explanation/libraries.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+There are a few things of note in this diagram:
+
+- `miral` and `mircore` are intended to be the primary user-facing libraries
+- `miroil` is a special compatibility layer for the Lomiri project is not
+  intended for general consumption
+- `mirserver` depends on the most libraries as it is the brains of the operation
+- `mircommon` provide common functionalities for every other library in the
+  system
+- The input and graphics platforms (represented as `input_X` and `graphics_X` in
+  the diagram) are dynamically loaded by `mirserver` when the engine starts up33333
```
Comment (RAOF) [inline doc/sphinx/explanation/libraries.md:52]:
```suggestion
  the diagram) are dynamically loaded by `mirserver` when the engine starts up.
```

**Review by RAOF — APPROVED** [review]
Typo aside, looks good to me.

**Pull request merged**